src/components/TextField/components/Resizer/Resizer.tsx

```diff
@@ -9,77 +9,72 @@ export interface ResizerProps {
-  private minimumLinesNode = React.createRef<HTMLDivElement>();
-  private animationFrame: number | null = null;
-
-  componentDidMount() {
-    this.handleHeightCheck();
-  }
-
-  componentDidUpdate() {
-    this.handleHeightCheck();
-  }
-
-  componentWillUnmount() {
-    if (this.animationFrame) {
-      cancelAnimationFrame(this.animationFrame);
-    }
-  }
-
-  render() {
-    const {contents, minimumLines} = this.props;
-
-    const minimumLinesMarkup = minimumLines ? (
-      <div
-        testID="MinimumLines"
-        ref={this.minimumLinesNode}
-        className={styles.DummyInput}
-        dangerouslySetInnerHTML={{
-          __html: getContentsForMinimumLines(minimumLines),
-        }}
-      />
-    ) : null;
-
-    return (
-      <div testID="ResizerWrapper" aria-hidden className={styles.Resizer}>
-        <EventListener event="resize" handler={this.handleHeightCheck} />
-        <div
-          testID="ContentsNode"
-          ref={this.contentNode}
-          className={styles.DummyInput}
-          dangerouslySetInnerHTML={{__html: getFinalContents(contents)}}
-        />
-        {minimumLinesMarkup}
-      </div>
-    );
-  }
-
-  private handleHeightCheck = () => {
-    if (this.animationFrame) {
-      cancelAnimationFrame(this.animationFrame);
+export function Resizer({
+  contents,
+  currentHeight,
+  minimumLines,
+  onHeightChange,
+}: ResizerProps) {
+  const contentNode = useRef<HTMLDivElement>(null);
+  const minimumLinesNode = useRef<HTMLDivElement>(null);
+  const animationFrame = useRef<number>();
+
+  useEffect(() => {
+    return () => {
+      if (animationFrame.current) {
+        cancelAnimationFrame(animationFrame.current);
+      }
+    };
+  }, []);
+
+  const minimumLinesMarkup = minimumLines ? (
+    <div
+      testID="MinimumLines"
+      ref={minimumLinesNode}
+      className={styles.DummyInput}
+      dangerouslySetInnerHTML={{
+        __html: getContentsForMinimumLines(minimumLines),
+      }}
+    />
+  ) : null;
+
+  const handleHeightCheck = useCallback(() => {
+    if (animationFrame.current) {
+      cancelAnimationFrame(animationFrame.current);
     }
 
-    this.animationFrame = requestAnimationFrame(() => {
-      const contentNode = this.contentNode.current;
-      const minimumLinesNode = this.minimumLinesNode.current;
-
-      if (!contentNode || !minimumLinesNode) {
+    animationFrame.current = requestAnimationFrame(() => {
+      if (!contentNode.current || !minimumLinesNode.current) {
         return;
       }
 
       const newHeight = Math.max(
-        contentNode.offsetHeight,
-        minimumLinesNode.offsetHeight,
+        contentNode.current.offsetHeight,
+        minimumLinesNode.current.offsetHeight,
       );
-      const {currentHeight, onHeightChange} = this.props;
 
       if (newHeight !== currentHeight) {
         onHeightChange(newHeight);
       }
     });
-  };
+  }, [currentHeight, onHeightChange]);
+
+  useLayoutEffect(() => {
```

Comment:
Need to use this so that the height gets adjusted before committing.

Reply:
`useLayoutEffect` doesn't play well SSR, we'll probably want to rethink this -> https://gist.github.com/gaearon/e7d97cdf38a2907924ea12e4ebdf3c85

Reply:
I opted for his second option since `useEffect` didn't work for us.